internal/github/github.go

```diff
@@ -0,0 +1,205 @@
+package github
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"net/http"
+	"os"
+	"strconv"
+
+	"github.com/Devleaps/github-secret-synchronizer/internal/vault"
+	"github.com/bradleyfalzon/ghinstallation/v2"
+	"github.com/google/go-github/v64/github"
+	"golang.org/x/crypto/nacl/box"
+)
+
+type GitHubWrapper struct {
+	client *github.Client
+
+	appID          int64
+	installationID int64
+	privateKey     []byte
+	orgName        string
+
+	publicKey string
+	keyID     string
+}
+
+// NewClient creates a new GitHub client
+func (g *GitHubWrapper) NewClient() error {
+	appID, installationID, privateKey, orgName, err := getGitHubAppCredentials()
+	if err != nil {
+		return err
+	}
+	g.appID = appID
+	g.installationID = installationID
+	g.privateKey = privateKey
+	g.orgName = orgName
+
+	itr, err := ghinstallation.New(http.DefaultTransport, g.appID, g.installationID, g.privateKey)
+	if err != nil {
+		return err
+	}
+
+	client := github.NewClient(&http.Client{Transport: itr})
+	g.client = client
+
+	publicKey, keyID, err := g.getOrgPublicKey()
+	if err != nil {
+		return err
+	}
+	g.publicKey = publicKey
+	g.keyID = keyID
+	return nil
+}
+
+// PushEncryptedSecret pushes the encrypted secret to GitHub
+func (g *GitHubWrapper) PushSecret(secret vault.VaultSecret) error {
+	encryptedSecret, err := encryptSecret(secret.Value, g.publicKey)
+	if err != nil {
+		return err
+	}
+	var ghSecret *github.EncryptedSecret
+	if secret.Visibility == vault.SELECTED_VISIBILITY {
+		repositoryIds, err := g.retrieveRepositoriesIds(secret.Repositories)
+		if err != nil {
+			return err
+		}
+		ghSecret = &github.EncryptedSecret{
+			Name:                  secret.Name,
+			KeyID:                 g.keyID,
+			EncryptedValue:        encryptedSecret,
+			Visibility:            secret.Visibility,
+			SelectedRepositoryIDs: repositoryIds,
+		}
+	} else {
+		ghSecret = &github.EncryptedSecret{
+			Name:           secret.Name,
+			KeyID:          g.keyID,
+			EncryptedValue: encryptedSecret,
+			Visibility:     secret.Visibility,
+		}
+	}
```

Comment:
```suggestion
         var selectedRepoIDs *SelectedRepoIDs
	if secret.Visibility == vault.SELECTED_VISIBILITY {
		repositoryIds, err := g.retrieveRepositoriesIds(secret.Repositories)
		if err != nil {
			return err
		}
		selectedRepoIDs = repositoryIds
	} else {
	  selectedRepoIDs = []	
	}
	ghSecret := &github.EncryptedSecret{
			Name:                  secret.Name,
			KeyID:                 g.keyID,
			EncryptedValue:        encryptedSecret,
			Visibility:            secret.Visibility,
			SelectedRepositoryIDs: selectedRepoIDs,
		}

```
something like this, less duplication

Reply:
I think I tried this before, but because `visibility` is not `selected`, GH API does not expect `SelectedRepositoryIDs` at all. But let me verify

Reply:
Found a different approach that works, great suggestion!